packages/lines/src/line/Line.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  BufferBindFlag,
+  BufferMesh,
+  BufferUsage,
+  Color,
+  MeshRenderer,
+  Script,
+  VertexElement,
+  VertexElementFormat,
+  Buffer,
+  IndexFormat,
+  Vector2
+} from "oasis-engine";
+import { LineMaterial } from "./material/LineMaterial";
+import { LineCap, LineJoin } from "./constants";
+import lineBuilder from "./vertexBuilder";
+
+export class Line extends Script {
+  protected _points: Vector2[] = [];
+  protected _cap = LineCap.Butt;
+  protected _join = LineJoin.Miter;
+  protected _renderer: MeshRenderer;
+  protected _material: LineMaterial;
+  protected _mesh: BufferMesh;
+  protected _needUpdate = false;
+  protected _flattenPoints: number[] = [];
+
+  /**
+   * The points that make up the line.
+   */
+  get points() {
+    return this._points;
+  }
+
+  set points(value: Vector2[]) {
+    this._points = value;
+    this._flattenPoints = this._points
+      .map((point) => {
+        return [point.x, point.y];
+      })
+      .flat();
+    this._needUpdate = true;
+  }
+
+  /**
+   * Determines the shape used to draw the end points of line.
+   */
+  get cap() {
+    return this._cap;
+  }
+
+  set cap(value: LineCap) {
+    if (value !== this._cap) {
+      this._cap = value;
+      this._material.cap = value;
+      this._needUpdate = true;
+    }
+  }
+
+  /**
+   * Determines the shape used to join two line segments where they meet.
+   */
+  get join() {
+    return this._join;
+  }
+
+  set join(value: LineJoin) {
+    if (value !== this._join) {
+      this._join = value;
+      this._material.join = value;
+      this._needUpdate = true;
+    }
+  }
+
+  /**
+   * The thickness of line.
+   */
+  get width() {
+    return this._material.width;
+  }
+
+  set width(value) {
+    this._material.width = value;
+  }
+
+  /**
+   * The color of line.
+   */
+  get color() {
+    return this._material.color;
+  }
+
+  set color(value) {
+    this._material.color = value;
+  }
+
+  constructor(entity) {
+    super(entity);
+  }
+
+  /**
+   * @override
+   */
+  onAwake(): void {
```

Comment:
If you don't want developer call this method, should add @internal

Reply:
fixed